File: backend/app/models/system.py
from __future__ import annotations

from pydantic import BaseModel


class SystemInfo(BaseModel):
    username: str
    hostname: str
    cwd: str
    project_version: str
    python_version: str
    scheduler: str
    workspace_root: str
    max_upload_size_mb: int


class SystemIdentity(BaseModel):
    app: str
    project_version: str
    pid: int
    scheduler: str
    workspace_root: str

File: frontend/src/api/system.ts
import { request } from './http'

export interface SystemInfo {
  username: string
  hostname: string
  cwd: string
  project_version: string
  python_version: string
  scheduler: string
  workspace_root: string
  max_upload_size_mb: number
}

export function getSystemInfo() {
  return request<SystemInfo>('/api/system/info')
}
